src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
+
+  static addAliveAnimals(item) {
+    Animal.alive.push(item);
+  }
+
+  static removeAnimals() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    Animal.addAliveAnimals(this);
+  }
+
+  hide() {
```

Comment:
The `hide` method should only set `hidden` to `true` according to the task description. If there is a need for a method to toggle the `hidden` state, it should be named differently and be a separate method.